### PR TITLE
Add matching pool token

### LINKED ISSUE #63: Separate Donation Tokens Used In Round
Currently, a `GrantRound` has one `donationToken`, which is used for both the matching pool contributions and individual grant contributions. We want to separate these such that the entity who creates the `GrantRound` can specify a different token used for each case:

`donationToken` for individual contributions
`matchingToken` for the matching pool contributions

Both should be provided in the `GrantRound` constructor. The `GrantRoundManager` should continue to hold the `donationToken` fixed for all Grants deployed through it, but should allow the `matchingToken` to be passed as a parameter.

This PR is dependent on #61.

### PR DESCRIPTION
**Pull request by corydickson**
Closes #63

### CODE REVIEW

File: contracts/contracts/GrantRoundManager.sol [hunk 1]
```diff
@@ -74,17 +75,20 @@ contract GrantRoundManager {
...
     address _owner,
     address _payoutAdmin,
+    IERC20 _matchingToken,
     GrantRegistry _registry,
     uint256 _startTime,
     uint256 _endTime,
     string memory _metaPtr,
     uint256 _minContribution
   ) external {
+    require(_matchingToken.totalSupply() > 0, "GrantRoundManager: Invalid matching token");
```
Comment (corydickson) [inline contracts/contracts/GrantRoundManager.sol:85]: Adding similar check to the donation token supply check in this contracts constructor

File: contracts/contracts/GrantRound.sol [hunk 2]
```diff
@@ -24,9 +24,12 @@ contract GrantRound {
   /// @notice GrantsRegistry
   GrantRegistry public immutable registry;
 
-  /// @notice ERC20 token that accepts pool donations
+  /// @notice ERC20 token that accepts donations
```
Comment (mds1) [inline contracts/contracts/GrantRound.sol:27]: I think this comment for the `donationToken` could be a bit more clear. Maybe

This comment a bit unclear
```suggestion
  /// @notice Token used for all contributions. Contributions in a different token are swapped to this token
```

File: contracts/contracts/GrantRound.sol [hunk 3]
```diff
@@ -24,9 +24,12 @@ contract GrantRound {
...
   IERC20 public immutable donationToken;
 
+  /// @notice ERC20 token that for the matching pool
```
Comment (mds1) [inline contracts/contracts/GrantRound.sol:30]: Similar, could probably be more explicit here
```suggestion
  /// @notice Token used to payout match amounts at the end of a round
```

File: contracts/contracts/GrantRound.sol [hunk 4]
```diff
@@ -52,6 +55,7 @@ contract GrantRound {
    * @param _payoutAdmin Grant round administrator that has permission to payout the matching pool
    * @param _registry Address that contains the grant metadata
    * @param _donationToken Address of the ERC20 token in which donations are made
+   * @param _donationToken Address of the ERC20 token for accepting matching pool contributions
```
Comment (mds1) [inline contracts/contracts/GrantRound.sol:58]: Typo. Also would update the `_donationToken` and `_matchingToken` comments to match any changes made to their state variable comments above

```suggestion
   * @param _matchingToken Address of the ERC20 token for accepting matching pool contributions
```

File: contracts/contracts/GrantRound.sol [hunk 5]
```diff
@@ -85,10 +92,10 @@ contract GrantRound {
 
   /**
    * @notice Before the round ends this method accepts matching pool funds
-   * @param _amount The amount of donation token that can be sent to the contract for the matching pool
+   * @param _amount The amount of matching token that can be sent to the contract for the matching pool
```
Comment (mds1) [inline contracts/contracts/GrantRound.sol:95]: Nit
```suggestion
   * @param _amount The amount of matching token that will be sent to the contract for the matching pool
```

File: contracts/contracts/GrantRoundManager.sol [hunk 6]
```diff
@@ -65,6 +65,7 @@ contract GrantRoundManager {
    * @notice Creates a new GrantRound
    * @param _owner Grant round owner that has permission to update the metadata pointer
    * @param _payoutAdmin Grant round administrator that has permission to payout the matching pool
+   * @param _matchingToken Address for the matching pool tokens
```
Comment (mds1) [inline contracts/contracts/GrantRoundManager.sol:68]: If we update the comment for this param above, just making a note to remember to do that here also

**Review by mds1 — CHANGES_REQUESTED** [review]
Looks great overall! My only comments are just on clarifying some of the docstrings/comments in the code to be a bit more explicit

File: contracts/contracts/GrantRoundManager.sol (hunk 6 above)
Comment (corydickson) [reply contracts/contracts/GrantRoundManager.sol]: Addressed [17133de ](https://github.com/dcgtc/dgrants/pull/85/commits/17133de9d645a163a0ebc57efa019b2745e10db3)

**Pull request merged**